test/suites/drift/suite_test.go

```diff
@@ -57,73 +69,653 @@ var _ = AfterEach(func() { env.Cleanup() })
 var _ = AfterEach(func() { env.AfterEach() })
 
 var _ = Describe("Drift", func() {
-
-	var pod *corev1.Pod
-
+	var dep *appsv1.Deployment
+	var selector labels.Selector
+	var numPods int
 	BeforeEach(func() {
-		env.ExpectSettingsOverridden(corev1.EnvVar{Name: "FEATURE_GATES", Value: "Drift=true"})
-
-		coretest.ReplaceRequirements(nodePool, karpv1.NodeSelectorRequirementWithMinValues{
-			NodeSelectorRequirement: corev1.NodeSelectorRequirement{
-				Key:      corev1.LabelInstanceTypeStable,
-				Operator: corev1.NodeSelectorOpIn,
-				Values:   []string{"Standard_DS2_v2"},
-			}})
-
-		// Add a do-not-disrupt pod so that we can check node metadata before we disrupt
-		pod = coretest.Pod(coretest.PodOptions{
-			ObjectMeta: metav1.ObjectMeta{
-				Annotations: map[string]string{
-					karpv1.DoNotDisruptAnnotationKey: "true",
+		numPods = 1
+		// Add pods with a do-not-disrupt annotation so that we can check node metadata before we disrupt
+		dep = coretest.Deployment(coretest.DeploymentOptions{
+			Replicas: int32(numPods),
+			PodOptions: coretest.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "my-app",
+					},
+					Annotations: map[string]string{
+						karpv1.DoNotDisruptAnnotationKey: "true",
+					},
 				},
+				TerminationGracePeriodSeconds: lo.ToPtr[int64](0),
 			},
-			ResourceRequirements: corev1.ResourceRequirements{Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("0.5")}},
-			Image:                "mcr.microsoft.com/oss/kubernetes/pause:3.6",
```

Comment:
I'm pretty sure if we don't supply this it will use an aws image. Feel we should be using ours?

Might be good to indirectly route through a wrapper package that sets this.